session.h

```diff
@@ -32,14 +32,23 @@ typedef struct {
 } session_t;
  /* TODO: Add support for RIOT over sockets  */
 #elif defined(WITH_RIOT_SOCK)
+#include "net/ipv4/addr.h"
 #include "net/ipv6/addr.h"
 typedef struct {
-  unsigned char size;     /**< size of session_t::addr */
+  unsigned char size;       /**< size of session_t::addr */
   struct {
-    unsigned short port;  /**< transport layer port */
-    ipv6_addr_t addr6;    /**< IPv6 address */
-  } addr;                 /**< session IP address and port */
-  int ifindex;            /**< network interface index */
+    unsigned short family;  /**< IP address family */
+    unsigned short port;    /**< transport layer port */
+    union {
+#ifdef SOCK_HAS_IPV4
+      ipv4_addr_t ipv4;     /**< IPv4 address */
+#endif
+#ifdef SOCK_HAS_IPV6
+      ipv6_addr_t ipv6;     /**< IPv6 address */
+#endif
+    };
```

Comment:
One potential issue I see here is that some (older) compilers do not allow anonymous unions. Unfortunately, I do not remember which one that was.

Reply:
I think this is less of an issue since this code is compiled only for RIOT OS and RIOT already uses anonymous unions. Mostly in driver code, but anonymous unions can also be found in parts of RIOT's primary network stack `gnrc`. So I would argue that compilers that do not allow anonymous unions are probably not considered anyway.

Reply:
Sounds reasonable.